tests/CMakeLists.txt

```diff
@@ -4,9 +4,15 @@ include(GoogleTest)
 add_library(test_common_libs INTERFACE)
 target_link_libraries(test_common_libs INTERFACE pthread gtest gtest_main)
 
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tt_metal/tt_metal)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tt_eager) # this should go away and be replaced with link to ttnn
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/ttnn/unit_tests/gtests)
 
-set(TESTS_DEPENDS_LIST metal_tests eager_tests unit_tests_ttnn unit_tests_ttnn_ccl test_multi_device galaxy_unit_tests_ttnn ttnn watcher_dump umd_tests)
-add_custom_target(tests DEPENDS ${TESTS_DEPENDS_LIST})
+option(TT_METAL_BUILD_TESTS "Enables build of tt_metal tests" OFF)
+if(TT_METAL_BUILD_TESTS)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tt_metal/tt_metal)
+endif(TT_METAL_BUILD_TESTS)
+
+option(TTNN_BUILD_TESTS "Enables build of ttnn tests" OFF)
+if(TTNN_BUILD_TESTS)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tt_eager) # this should go away and be replaced with link to ttnn
```

Comment:
@vtangTT whats the deal with this comment?

Is now a good time to fix this?

Why does tests/ directory have both tt_eager and ttnn ?